libsupport/src/ArrowVisitor.cpp

```diff
@@ -0,0 +1,103 @@
+#include "katana/ArrowVisitor.h"
+
+#include <arrow/type_traits.h>
+
+#include "katana/Logging.h"
+
+struct ToArrayVisitor {
+  // Internal data and constructor
+  const std::vector<std::shared_ptr<arrow::Scalar>>& scalars;
+  ToArrayVisitor(const std::vector<std::shared_ptr<arrow::Scalar>>& input)
+      : scalars(input) {}
+
+  using ReturnType = std::shared_ptr<arrow::Array>;
+  using ResultType = katana::Result<ReturnType>;
+  template <typename ArrowType, typename BuilderType>
+  arrow::enable_if_t<
+      arrow::is_number_type<ArrowType>::value ||
+          arrow::is_boolean_type<ArrowType>::value ||
+          arrow::is_temporal_type<ArrowType>::value,
+      ResultType>
+  Call(BuilderType* builder) {
+    using ScalarType = typename arrow::TypeTraits<ArrowType>::ScalarType;
+
+    if (auto st = builder->Resize(scalars.size()); !st.ok()) {
```

Comment:
Weird note, since the builder always has 0 elements coming in here, `Resize` is ok. If there were one or more elements in the builder coming in, `Resize` would lead to a segfault on the last appends. Preference for `Reserve`, but because it is only used inside this file, should be fine either way

Reply:
I read the documentation on `Resize `and `Reserve `like 4 times and apparently still made the wrong choice.  Fixed.